### PR TITLE
Update OTLP Exporter README

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Remove note about `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` not being supported. This was completed in #4667

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md [hunk 1]
```diff
@@ -175,8 +175,6 @@ values of the `PeriodicExportingMetricReaderOptions`
 | `OTEL_METRIC_EXPORT_INTERVAL`                       | `ExportIntervalMilliseconds`                    |
 | `OTEL_METRIC_EXPORT_TIMEOUT`                        | `ExportTimeoutMilliseconds`                     |
 
-NOTE: `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE` is not supported yet [#3756](https://github.com/open-telemetry/opentelemetry-dotnet/issues/3756).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:178]: Just mention with one liner that the environment variable is supported.

**Review by vishweshbankwar — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:178]: please also call out the values suported. I dont think we support all the values. (like lowmemory)

Comment (utpilla) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:178]: What is `lowmemory`?

Comment (austindrenski) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:178]: @cijothomas is support for `LowMemory` on the roadmap, or blocked by anything specific we can follow in the meantime?

- https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/metrics/sdk_exporters/otlp.md#additional-configuration

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:178]: @utpilla The above link has the spec for `lowmemory`.
@austindrenski I don't think there are any blockers implementing lowmemory. Its just that nobody asked for it so far! Could you open an issue for it to better track adding it!

**Pull request merged**